lib/API/JSON/v1/Settings.php

```diff
@@ -104,10 +110,17 @@ public function set($settings = []) {
   public function recalculateSubscribersScore() {
     $this->statisticsOpensRepository->resetSubscribersScoreCalculation();
     $this->statisticsOpensRepository->resetSegmentsScoreCalculation();
-    $task = new ScheduledTaskEntity();
-    $task->setType(SubscribersEngagementScore::TASK_TYPE);
-    $task->setScheduledAt(Carbon::createFromTimestamp($this->wp->currentTime('timestamp')));
-    $task->setStatus(ScheduledTaskEntity::STATUS_SCHEDULED);
+    $task = $this->scheduledTasksRepository->findOneBy([
+      'type' => SubscribersEngagementScore::TASK_TYPE,
+      'status' => ScheduledTaskEntity::STATUS_SCHEDULED,
+    ]);
+    if ($task) {
+      $task->setScheduledAt(Carbon::createFromTimestamp($this->wp->currentTime('timestamp')));
+    } else {
+      $task = new ScheduledTaskEntity();
+      $task->setType(SubscribersEngagementScore::TASK_TYPE);
+      $task->setStatus(ScheduledTaskEntity::STATUS_SCHEDULED);
+    }
```

Comment:
It looks like that a newly created task has not scheduledAt set. I guess it could be done the other way around.
```php
if (!$task) {
  $task = new ScheduledTaskEntity();
  $task->setType(SubscribersEngagementScore::TASK_TYPE);
  $task->setStatus(ScheduledTaskEntity::STATUS_SCHEDULED);
} 
$task->setScheduledAt(Carbon::createFromTimestamp($this->wp->currentTime('timestamp')));
```

Reply:
Right, it was my original idea, but I don't know why I didn't make it. Thank you